x/mongo/driver/operation.go

```diff
@@ -535,7 +525,7 @@ func (op Operation) Execute(ctx context.Context, scratch []byte) error {
 		if deadline, ok := ctx.Deadline(); ok {
 			if internal.IsTimeoutContext(ctx) && time.Now().Add(srvr.RTT90()).After(deadline) {
 				err = internal.WrapErrorf(ErrDeadlineWouldBeExceeded,
-					"Remaining timeout %v applied from Timeout is less than 90th percentile RTT", time.Until(deadline))
+					"remaining time %v until context deadline is less than 90th percentile RTT", time.Until(deadline))
```

Comment:
We should include the calculated 90th percentile RTT value here to make troubleshooting RTT issues easier in the future. Consider also including the number of RTT samples collected in the error, too.

For example, if we calculate 90th percentile RTT based on a single sample on startup, we may end up short-circuiting some requests due to an outlier RTT sample. If that happens, we want to have a record of the state of the RTT monitor so we understand the impact of short-circuiting requests based on RTT and the range of possible states of the RTT monitor.

Reply:
A great point. Working on improving this error message.

Reply:
I think I'll add a `Stats` method to the rtt monitor that will return relevant stats in these `ErrDeadlineWouldBeExceeded` errors. That will be a slightly larger change, as I'll have to modify the `Server` interface, as well. I will do that as part of GODRIVER-2339.